packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -460,11 +471,48 @@ abstract class AssetBundleImageProvider extends ImageProvider<AssetBundleImageKe
   ///
   /// This function is used by [load].
   @protected
-  Future<ui.Codec> _loadAsync(AssetBundleImageKey key) async {
+  Future<ui.Codec> _loadAsync(AssetBundleImageKey key, {Size size}) async {
     final ByteData data = await key.bundle.load(key.name);
     if (data == null)
       throw 'Unable to read data';
-    return await PaintingBinding.instance.instantiateImageCodec(data.buffer.asUint8List());
+    return await PaintingBinding.instance.instantiateImageCodec(
+        data.buffer.asUint8List(), size: size);
+  }
+}
+
+class _SizeAwareCacheKey {
+  const _SizeAwareCacheKey(this._providerCacheKey, this._imageSize);
+
+  final dynamic _providerCacheKey;
+  final Size _imageSize;
+}
+
+/// Re-sizes the image provided to the given size.
+///
+/// See also:
+///
+/// * [Image.network] for example usage when `resizeToFit` parameter is set.
+class ResizedImage
+    extends ImageProvider<_SizeAwareCacheKey> {
+  /// Creates an object that re-sizes the image to the given size.
+  ///
+  /// The arguments must not be null.
+  const ResizedImage(this._imageProvider, this._imageSize);
+
+  final ImageProvider _imageProvider;
+  final Size _imageSize;
+
+  @override
+  ImageStreamCompleter load(_SizeAwareCacheKey key, {Size size}) {
+    assert(key._imageSize == _imageSize);
+    return _imageProvider.load(key._providerCacheKey, size: _imageSize);
+  }
+
+  @override
+  Future<_SizeAwareCacheKey> obtainKey(
+      ImageConfiguration configuration) async {
```

Comment:
The ImageConfiguration already has the size information, we should find a way to use that